metrictank/src/main/java/com/expedia/metrics/metrictank/GaasKafkaValueDeserializer.java

```diff
@@ -0,0 +1,103 @@
+package com.expedia.metrics.metrictank;
+
+import com.expedia.metrics.MetricData;
+import com.expedia.metrics.MetricDefinition;
+import com.expedia.metrics.TagCollection;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class GaasKafkaValueDeserializer implements Deserializer {
+    private final static MessagePackSerializer mps = new MessagePackSerializer();
+
+    @Override
+    public void configure(Map configs, boolean isKey) {
+
+    }
+
+    @Override
+    public MetricData deserialize(String topic, byte[] data) {
+        try {
+
+            //Refer to
+            // 1 - https://github.com/raintank/schema/blob/master/msg/msg.go#L114
+            // 2 - https://github.com/raintank/schema/blob/master/msg/format.go
+            Format format = Format.values()[data[0]];
+
+            if (format == Format.FormatMetricDataArrayJson) {
+                return mps.deserialize(data);
+            }
+            else if (data.length == 33 && format == Format.FormatMetricPoint) {
+                return deserializeV2(Arrays.copyOfRange(data, 1, data.length));
+            }
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Not sure that we really want to do this in the case of an unrecognized message. I get the idea but I'm imagining a ton of metric data with some unknown format filling up the logs with stacktraces.